notebooks/en/constrained_generation.ipynb

```diff
@@ -0,0 +1,522 @@
+{
```

Comment:
<p>"A grammar" sounds a bit awkward</p><p> </p><p>"To force a JSON output, we'll have to use <strong>grammar</strong>.</p><p> </p><p><strong>Grammar</strong> is a set of rules..."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/64/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Reply:
<p>@stevhliu I used "a grammar" to stress that there can be many different grammars. Does your comment above say that even in indefinite form, it would be better to change the "a grammar" to "grammar" for a better style? To me saying "Grammar is a set of rules" seems to imply that Grammar is a unique noun, like Transformers or Hugging Face, when it should be indefinitee like "A LLM" (sorry to have to clarify, I'm not proficient enough with English to have this nuance)</p><p> </p>

Reply:
No need to apologize!

Grammar is not unique at all unless you specify it to be so, like *English grammar* or *German grammar*. So in this case, saying "Grammar is a set of rules" is still very general and understood to just be a set of rules for any language since we haven't specified it :)

Reply:
<p>Thank you, very interesting!</p>